node/src/command.rs

```diff
@@ -52,20 +51,29 @@ impl SubstrateCli for Cli {
     }
 
     fn load_spec(&self, id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-        Ok(match id {
-            "dev" => Box::new(chain_spec::development_config()),
-            "local" => Box::new(chain_spec::local_testnet_config()),
-            "" | "main" => Box::new(chain_spec::main_config()),
-            "arcadia" => Box::new(chain_spec::arcadia_config()),
-            "staking-local" => Box::new(chain_spec::local_staking_config()),
-            path => Box::new(chain_spec::ChainSpec::from_json_file(
-                std::path::PathBuf::from(path),
-            )?),
-        })
+        if self.run.runtime.is_staking_runtime() {
+            Ok(match id {
+                "dev" => Box::new(chain_spec::cs_staking::development_config()),
+                "" | "local" => Box::new(chain_spec::cs_staking::local_staking_config()),
+                path => Box::new(chain_spec::cs_staking::ChainSpec::from_json_file(
+                    std::path::PathBuf::from(path),
+                )?),
+            })
+        } else {
+            Ok(match id {
+                "dev" => Box::new(chain_spec::cs_main::development_config()),
+                "local" => Box::new(chain_spec::cs_main::local_testnet_config()),
+                "" | "main" => Box::new(chain_spec::cs_main::main_config()),
+                "arcadia" => Box::new(chain_spec::cs_main::arcadia_config()),
+                path => Box::new(chain_spec::cs_staking::ChainSpec::from_json_file(
+                    std::path::PathBuf::from(path),
+                )?),
+            })
+        }
```

Comment:
I think we should have a more standard setup where we define the following chain specs, without the runtime cli arg:
- `staking-dev`
- `staking-local`

Reply:
The reason behind introduction of `--runtime` cli, is to handle the case of loading the chainspec from file path. The type `ChainSpec` is runtime specific, & this extra option is used to distinguish between the runtime to choose.

Reply:
Code refactor done, chain is identified from chain spec ID. introduced trait `IdentifyChain` to handle it :+1: